Source/WebCore/rendering/RenderBlockFlow.cpp

```diff
@@ -746,12 +746,15 @@ void TextBoxTrimmer::adjustTextBoxTrimAfterLayout()
 
 void RenderBlockFlow::layoutInFlowChildren(bool relayoutChildren, LayoutUnit& repaintLogicalTop, LayoutUnit& repaintLogicalBottom, LayoutUnit& maxFloatLogicalBottom)
 {
-    if (!firstChild()) {
+    if (!firstChild() || (firstChild()->isSkippedContentForLayout() && !normalChildNeedsLayout())) {
```

Comment:
Same question here: can you elaborate why these changes are needed? Are the changes in `layoutIfNeeded` not good enough to skip work here?

Reply:
This change is a bit vague indeed, it improves performance but I will check tomorrow if there is a more logical place that gives the same performance.